Framework/src/QualitySender.cxx

```diff
@@ -0,0 +1,69 @@
+#include "QualityControl/QualitySender.h"
+#include <Framework/DataRefUtils.h>
+#include <Framework/InputRecordWalker.h>
+#include "QualityControl/QualitiesToFlagCollectionConverter.h"
+#include "QualityControl/QualityObject.h"
+
+namespace o2::quality_control::flag_sender
+{
+
+QualitySender::QualitySender(const std::string& grpcUri) : mGrpcUri{ grpcUri } {}
+
+void QualitySender::run(framework::ProcessingContext& context)
+{
+  for (auto const& ref : framework::InputRecordWalker(context.inputs())) {
+    try {
+      mQualityObjects.push_back(std::move(framework::DataRefUtils::as<core::QualityObject>(ref)));
+    } catch (...) {
+    }
+  }
+}
+
+auto merge(std::unique_ptr<QualityControlFlagCollection> collection, std::shared_ptr<core::QualityObject> qualityObject) -> std::unique_ptr<QualityControlFlagCollection>
+{
+  core::QualitiesToFlagCollectionConverter converter(std::move(collection), qualityObject->getPath());
+  converter(*qualityObject);
+  return converter.getResult();
+}
+
+auto collectionFromQualityObject(const core::QualityObject& qualityObject) -> std::unique_ptr<QualityControlFlagCollection>
+{
+  return std::make_unique<QualityControlFlagCollection>(
+    qualityObject.getName(),
+    qualityObject.getDetectorName(),
+    core::gFullValidityInterval, // we don't need exact run duration because each flag have it's own range.
```

Comment:
Yes and no. If a part of run does not get any flag, such timespan is marked flagged `UnknownQuality`: https://github.com/AliceO2Group/QualityControl/blob/cc46357fceebcf8f0e00289e8a96d06756122f00/Framework/src/QualitiesToFlagCollectionConverter.cxx#L82

Thus we would need to make sure with BKP devs that e.g. sending them a flag `UnknownQuality` spanning from time `0` to an arbitrary time during the run will be handled by BKP correctly.